python/task_manager/background_task_manager.py

```diff
@@ -325,15 +429,13 @@ def _get_worker_thread(self):
         # create a new worker thread - note, there are two different implementations of the WorkerThread class
         # that use two different recipes.  Although WorkerThreadB is arguably more correct it has some issues
         # in PyQt so WorkerThread is currently preferred - see the notes in the class above for further details
-        thread = WorkerThread(self)
+        thread = WorkerThread(self._results_poller)
         if not isinstance(thread, WorkerThread):
             # for some reason (probably memory corruption somewhere else) I've occasionally seen the above
             # creation of a worker thread return another arbitrary object!  Added this in here so the code
             # will at least continue correctly and not do unexpected things!
             self._bundle.log_error("Failed to create background worker thread for task Manager!")
             return None
-        thread.task_failed.connect(self._on_worker_thread_task_failed)
```

Comment:
We no longer connect to the threads, the poller will instead call those callbacks.